code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -444,18 +470,14 @@ GLOBAL_VAR(medibot_unique_id_gen)
 	if(declare_crit && C.health <= 0) //Critical condition! Call for help!
 		declare(C)
 
-	//They're injured enough for it!
-	if(C.getBruteLoss() >= heal_threshold)
-		return TRUE //If they're already medicated don't bother!
-
-	if(C.getOxyLoss() >= (5 + heal_threshold))
-		return TRUE
-
-	if(C.getFireLoss() >= heal_threshold)
-		return TRUE
-
-	if(C.getToxLoss() >= heal_threshold)
-		return TRUE
+	//If they're injured, we're using a beaker
+	var/S = FALSE
+	if((reagent_glass) && ((C.maxHealth - C.health >= heal_threshold)))
+		S = TRUE
+		for(var/datum/reagent/R in reagent_glass.reagents.reagent_list)
+			if(C.reagents.has_reagent(R.type) || check_overdose(C, R.type, injection_amount))
```

Comment:
```suggestion
			if(C.reagents.has_reagent(R.type))
```

I don't believe overdose check shouldn't be present unless there is a way to bypass it via disabling safeties / emagging.

Reply:
I don't seem to understand, you mean it shouldn't check for overdose? if its emagged it already passes the check as above there is an if(emagged) return true

Reply:
The former is as easy as removing the check. The latter is somewhat better and more newbie friendly but takes more work.

The current setup makes it require a toxic or outright harmful reagent to sabotage.

Reply:
Feel free to resolve this if you don't feel like doing it - like I said, this isn't a requirement but I'd like your side one way or the other before I start pushing for Dragon to weigh in and potentially merge this over the weekend.